### DISCUSSION TITLE
Welcome to WordPressPCL Discussions!

### DISCUSSION THREAD

**Discussion by ThomasPe** [wp-net/WordPressPCL#238] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by brendonjdev** [comment]
Hi All,

I need some advice to choose my development path/language based on the following, and how much the PCL can do.

I love programming in C#, and I'm about to learn Xamarin, but what I need to accomplish, depends on whether I have to go to REACT native. I use a Wordpress App plugin known as BuddyBoss. It is a monster plugin and major upgrade from Buddypress. It is the most advanced Social Media Community (mini facebook) "application" out there. (If anyone knows of something better - Let me know please.)

I have to create a Native Mobile App though that will provide the same functionality that the site has on a webpage. "BuddyBoss" (The developer of the module) has two options when it comes to creating a mobile app from the wordpress site. 1 - "We'll do it for you." (Costs a fortune). 2 - I"f you are a React developer, you will easily be able to create the Mobile App yourself" (Free).

But because C# is my coding language, I would way more prefer to build the Mobile app in Xamarin, and use PCL to interact with the Wordpress site. 

Question: Will I be able to develop the Xamarin App, using the wordpress REST Api plugin and the Wordpress/C# CPL plugin, and still have the same advanced functionality and features that the wordpress website has when accessing it via a browser? 

Basic Features I'm referring are the following:

Private messaging (Including sharing media)
Following
Friend requests
Instant notification of new messages etc...

Advanced profile type creation and editing
Their own blog
Their own media albums/directory

And obviously the bells and whistles like displaying their IG posts on their profiles etc...

Group and Forum creation by Members - And then full member interaction in those groups and forums.

Activity feed of absolutely everything members do - The members can select what they want to see. 


So - Will I at least be able to do all of that with Xamarin with WordPress Rest API and WordpressPCL, or would it be better to go with REACT Native?

I look forward to your answers and feedback. All feedback and suggestions are welcome!

Thanks,

Brendon
